src/lit-html.ts

```diff
@@ -598,19 +596,14 @@ export class TemplateInstance {
           null as any,
           false);
 
-      const parts = this.template.parts;
-      let index = 0;
-      let partIndex = 0;
-      let templatePart = parts[0];
-      let node = walker.nextNode();
-      while (node != null && partIndex < parts.length) {
-        if (index === templatePart.index) {
-          this._parts.push(this._partCallback(this, templatePart, node));
-          templatePart = parts[++partIndex];
-        } else {
+      let index = -1;
+      for (let i = 0; i < parts.length; i++) {
+        const part = parts[i];
+        while (index < part.index) {
```

Comment:
`while (index++ < part.index) {` should work also

Reply:
Can't do that, since it'd still increment `index` when `part.index` already matched.

```js
html`<div first=${thisWouldWork} second=${butNowIndexIncremented} third=${andICreateAnInfiniteLoop}></div>`

Reply:
Ah, good point